From b0e29fa3cb2a0167e78bc95cccd3f90cd7ce5c33 Mon Sep 17 00:00:00 2001
From: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
Date: Wed, 14 Sep 2022 14:40:55 +0800
Subject: [PATCH 13/28] dt-binding: Add binding file for SGI mailbox

Add device tree binding file for the new added mailbox device.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
---
 .../bindings/mailbox/arm,sgi-mbox.yaml        | 58 +++++++++++++++++++
 1 file changed, 58 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/mailbox/arm,sgi-mbox.yaml

diff --git a/Documentation/devicetree/bindings/mailbox/arm,sgi-mbox.yaml b/Documentation/devicetree/bindings/mailbox/arm,sgi-mbox.yaml
new file mode 100644
index 000000000000..70354954fa07
--- /dev/null
+++ b/Documentation/devicetree/bindings/mailbox/arm,sgi-mbox.yaml
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/mailbox/arm,sgi-mbox.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: ARM SGI Mailbox
+
+maintainers:
+  - Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
+
+description: |
+  The SGI Mailbox is a virtual device, which uses SGI interrupt to
+  notify remote side and share memory for device MMIO registers.
+
+properties:
+  compatible:
+    oneOf:
+      - const: fsl,arm-sgi-mbox
+  reg:
+    maxItems: 1
+
+  "#mbox-cells":
+    description: |
+      <&phandle type channel>
+      phandle   : Label name of controller
+      type      : Channel type
+      channel   : Channel index
+
+      This mailbox support 3 type of unidirectional channels, each type
+      has 4 channels. A total of 12 channels. Following types are
+      supported:
+      0 - TX channel with 32bit transmit register
+      1 - RX channel with 32bit receive register and IRQ support
+      2 - RX doorbell channel.
+    const: 2
+
+required:
+  - compatible
+  - reg
+  - "#mbox-cells"
+
+additionalProperties: false
+
+examples:
+  - |
+    reserved-memory {
+        sgi-mbox@b8500000 {
+            no-map;
+            reg = <0 0xb8500000 0 0x1000>;
+        };
+    };
+
+    sgi-mailbox@b8500000 {
+        compatible = "fsl,arm-sgi-mbox";
+        reg = <0 0xb8500000 0 0x1000>;
+        #mbox-cells = <2>;
+    };
-- 
2.34.1

